MagicBytesValidator.Tests/ValidatorIsValidAsync.cs

```diff
@@ -11,61 +11,39 @@
 
 namespace MagicBytesValidator.Tests;
 
-public class IsValidTests
+public class ValidatorIsValidAsync
 {
     private readonly Validator _validator;
-    private readonly FileType _fileTypeGif;
-    private readonly FileType _fileTypePng;
-    private readonly FileType _fileTypeZip;
+    private readonly IFileType _fileTypeGif;
+    private readonly IFileType _fileTypePng;
     private readonly MemoryStream _gifMemoryStream;
-    private readonly MemoryStream _zipMemoryStream;
-    private readonly Gif _gif;
-    private readonly Png _png;
-    private readonly Zip _zip;
+    private readonly MemoryStream _pngMemoryStream;
 
-    public IsValidTests()
+    public ValidatorIsValidAsync()
     {
-        _gif = new Gif();
-        _png = new Png();
-        _zip = new Zip();
-        _gifMemoryStream = new MemoryStream();
-        _zipMemoryStream = new MemoryStream();
+        var gif = new Gif();
+        var png = new Png();
+        _gifMemoryStream = new MemoryStream(gif.MagicByteSequences[0]);
+        _pngMemoryStream = new MemoryStream(png.MagicByteSequences[0]);
 
         _validator = new Validator();
-        _fileTypeGif = _validator.Mapping.FindByExtension(_gif.Extensions[0]) ?? throw new NullReferenceException();
-        _fileTypePng = _validator.Mapping.FindByMimeType(_png.MimeTypes.First()) ?? throw new NullReferenceException();
-        _fileTypeZip = _validator.Mapping.FindByMimeType(_zip.MimeTypes.First()) ?? throw new NullReferenceException();
+        _fileTypeGif = _validator.Mapping.FindByExtension(gif.Extensions[0]) ?? throw new NullReferenceException();
```

Comment:
Null check needed?

Reply:
What do you mean? That this check is unnecessary?